.github/workflows/ci.yml

```diff
@@ -0,0 +1,60 @@
+name: CI Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    env:
+      TZ: Europe/Oslo
+    steps:
+      - uses: actions/checkout@v3
+
+      # nodejs is needed because the dynamic download of it via the prettier maven plugin often
+      # times out
+      # Example: https://github.com/opentripplanner/OpenTripPlanner/actions/runs/4490450225/jobs/7897533439
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          cache: maven
+
+      - name: Run tests
+        run: |
+          sudo timedatectl set-timezone "Europe/Oslo"
+          mvn --batch-mode jacoco:prepare-agent test jacoco:report -Dprettier.nodePath=node -Dprettier.npmPath=npm
+
+  image:
+    # only build and push the image when it's a commit to master in Entur's repository
+    if: github.repository_owner == 'entur' && github.event_name == 'push'
```

Comment:
This will push (after building) the Docker image on *every branch*. ⚠️
I propose to implement it as [I did in my PR](https://github.com/entur/lamassu/pull/214/commits/f4c30e138e6d2d28d89dcfeae07865b7fbab35e6#diff-9136d7ee194d90376becf626f92179e6d99617d858ce125f6f73ca09fa410840R34): push only if on the main branch (`master`), but always build the Docker image to make errors in the Dockerfile visible.

Reply:
It actually only runs on master branch, because of this:

```
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```

So only push events on master are considered triggers.

Reply:
Yes, but this way, it doesn't validate PRs, so one can accidentally merge a broken `Dockerfile`.

Reply:
True I'll address this in #249 as well, since this PR is now closed :)